examples/kelvin_helmholtz_instability.jl

```diff
@@ -0,0 +1,366 @@
+                              advection = WENO5(),
+                                   grid = grid,
+                               coriolis = nothing,
+                      background_fields = (u=U, b=B),
+                                closure = IsotropicDiffusivity(ν=1e-4, κ=1e-4),
+                               buoyancy = BuoyancyTracer(),
+                                tracers = :b)
+
+# # A _Power_ful algorithm
+#
+# We will find the most unstable mode of the Kelvin-Helmholtz instability using the "power method". 
+# Linear instabilities, such as the Kelvin-Helmholtz instability, are described by linearized equations about a base state that take the form
+# ```math
+# \partial_t \Phi = L \Phi \, .
+# ```
+# In this example, ``\Phi = (u, v, w, b)`` is a vector of the velocities ``u, v, w``, and 
+# buoyancy ``b`` and ``L`` is a linear operator that depends on the base state (the `background_fields`) and other problem parameters.
+#
+# The linear operator ``L`` has eigenmodes ``u_j`` and corresponding and corresponding 
+# eigenvalues ``\lambda_j``,
+# ```math
+# L \, \phi_j = \lambda \, \phi_j \quad j=1,2,\dots \, .
+# ```
+# We use the convention that eigenvalues are ordered according to their real part, ``\real(\lambda_1) \ge \real(\lambda_2) \ge \dotsb``.
+# 
+# Successive application of ``L`` to a random initial state will render it parallel 
+# with eigenmode ``\phi_1``:
+# ```math
+# \lim_{n \to \infty} L^n \Phi \propto \phi_1 \, .
+# ```
+# Of course, if ``\phi_1`` is an unstable mode, i.e., its eigenvalue has ``\sigma_1 = \real(\lambda_1) > 0``, then successive application 
+# of ``L`` will lead to exponential amplification. (Or, if ``\sigma_1 < 0``, it will
+# lead to exponential decay of ``\Phi`` down to machine precision.) Therefore, after each 
+# application of the linear operator ``L`` to our state ``\Phi``, we should rescale the output ``L \Phi `` back to a pre-selected amplitude before applying ``L`` again.
+# 
+# Oceananigans.jl, does not include a "linear" version of the equations, but we can ensure we 
+# remain in the "linear" regime if we pick the pre-selected amplitude to be small enough, i.e.,
+# ensuring that terms quadratic in perturbations are much smaller than all other terms.
+# 
+# So, we initialize a `simulation` with random initial conditions with amplitude much less than those of
+# the base state (which are ``O(1)``). Each iteration of the power method includes:
+# - compute the perturbation energy, ``E_0``,
+# - evolve the system for ``\Delta t``,
+# - compute the perturbation energy, ``E_1``,
+# - determine the exponential growth of the most unstable mode during interval ``\Delta t`` as  ``\log(E_1 / E_0) / (2 \Delta t)``,
+# - repeat the above until growth rate converges.
+# 
+# By fiddling a bit with ``\Delta t`` we can get convergence after only a few iterations.
+# 
+# For this example, we take ``\Delta t = 20``.
+
+simulation = Simulation(model, Δt=0.1, iteration_interval=20, stop_iteration=100)
```

Comment:
It's a bit unfortunate that in the line above we talk about the time-interval `Δt` for each iteration of the power method and then we see `Δt=0.1` to set the time-step.

Perhaps we can call the former `Δτ`?

Reply:
done in e459279